pygeo/parameterization/DVGeoCST.py

```diff
@@ -0,0 +1,1175 @@
+            print(f"    Fit CST coefficients: {self.defaultDV[dvType]}")
+
+    def addPointSet(self, points, ptName, boundTol=1e-10, **kwargs):
+        """
+        Add a set of coordinates to DVGeometry
+        The is the main way that geometry in the form of a coordinate list is given to DVGeometry
+        to be manipulated.
+
+        Parameters
+        ----------
+        points : array, size (N,3)
+            The coordinates to embed.
+        ptName : str
+            A user supplied name to associate with the set of coordinates.
+            This name will need to be provided when updating the coordinates or when
+            getting the derivatives of the coordinates.
+        boundTol : float, optional
+            Small absolute deviation by which the airfoil coordinates can exceed the initial
+            minimum and maximum x coordinates, by default 1e-10.
+        kwargs
+            Any other parameters are ignored.
+        """
+        # Convert points to the type specified at initialization (with isComplex) and store the points
+        points = points.astype(self.dtype)
+
+        # Check that all points are within the airfoil x bounds
+        if np.any(points[:, self.xIdx] < self.xMin - boundTol) or np.any(points[:, self.xIdx] > self.xMax + boundTol):
+            raise ValueError(
+                f'Points in the point set "{ptName}" have x coordinates outside'
+                + f"the min and max x values in the initial dat file ({self.xMin} and {self.xMax})"
+            )
+        self.updated[ptName] = False
+        self.points[ptName] = {
+            "points": points,
+            "xMax": self.xMax.copy(),
+            "xMin": self.xMin.copy(),
+            "thicknessTE": self.thicknessTE.copy(),
+        }
+
+        # Determine which points are on the upper and lower surfaces
+        self.points[ptName]["upper"], self.points[ptName]["lower"] = self._splitUpperLower(points)
+
+        # If debug mode is on, plot the upper and lower surface points
+        if self.debug:
+            fig = plt.figure()
+            plt.scatter(
+                self.points[ptName]["points"][:, self.xIdx][self.points[ptName]["upper"]],
+                self.points[ptName]["points"][:, self.yIdx][self.points[ptName]["upper"]],
+                c="b",
+            )
+            plt.scatter(
+                self.points[ptName]["points"][:, self.xIdx][self.points[ptName]["lower"]],
+                self.points[ptName]["points"][:, self.yIdx][self.points[ptName]["lower"]],
+                c="r",
+            )
+            plt.scatter(
+                self.points[ptName]["points"][:, self.xIdx],
+                self.points[ptName]["points"][:, self.yIdx],
+                s=3,
+                c="k",
+                zorder=3,
+            )
+            plt.legend(["Upper", "Lower"])
+            plt.show()
+            plt.close(fig)
+
+    def addDV(self, dvName, dvType, lower=None, upper=None, scale=1.0, default=None):
```

Comment:
I just realized this; isnt it confusing that the DV names can get upper and lower and there are also upper and lower bounds on all dvs? maybe we change the upper and lower dvs to upper_coef and lower_coef? any name other than the upper/lower naming convention is probably fine